blog/_posts/2018-06-10-bsp.md

```diff
@@ -0,0 +1,214 @@
+alongside LSP.
+
+## The Build Server Protocol v1.0.0
+
+Today I announce version `v1.0.0` of the Build Server Protocol. This version
+has been the result of a fruitful collaboration with [Justin
+Kaeser][jastice], the [JetBrains] developer responsible for sbt support in
+[IntelliJ IDEA].
+
+Justin and I have spent the past months implementing a prototype of the Build
+Server Protocol between [IntelliJ IDEA] (client) and [Bloop] (server). The
+goal of this prototype was two-fold. On the one hand, we wanted to experience
+first-hand the challenges of implementing BSP and learn new insights to
+improve the protocol. On the other hand, we wanted to provide Scala
+developers a faster "Import Project" that would work across different build
+tools *now*, even though build tools don't yet provide implementations of
+build servers. Thanks to Bloop, users of [sbt], [mill] or [Maven] ---build
+tools that Bloop currently supports--- can try this prototype out.
+
+[BSP v1.0.0](https://github.com/scalacenter/bsp/blob/master/docs/bsp.md)
+has been designed to support the most basic language integration in an
+IDE/editor. IDEs can find all the modules defined in a workspace, ask for
+their dependencies, compile, test, run, and get notifications on modified
+build targets, among a few other features.
+
+Our IntelliJ-Bloop implementation doesn't yet implement all these endpoints.
+Instead, it focuses on providing a faster integration than the traditional
+project import. You can now import your project, browse Scala code with full
+language support and get compiler diagnostics from the build tool in
+IntelliJ's editor.
+
+How does this work? First, you export your project from your build to Bloop.
+This process takes usually ~10s for a medium-sized project (highly
+recommended to use [coursier] in your build for upmost speed) and exists so
+that Bloop can understand different build tools. Then, the IDE imports it
+from Bloop via BSP. In total, the "Import Project" process takes around ~15
+seconds.
+
+You can see it on this demo importing [akka/akka](https://github.com/akka/akka/):
+
+![Demo of BSP import for akka/akka](/resources/img/bloop-install.gif)
+
+In my experience, importing a build via BSP is substantially faster than
+doing so via the traditional sbt or Maven project import where the process
+can take minutes. It also enables smarter communication between build tools
+and IDEs because the communication is bidirectional.
+
+Further, this two-step process can be improved in the future. First, bloop
+could run `bloopInstall` every time you start up your sbt shell or `reload`
+your build. Second, Bloop's Build Server could send notifications the IDE
+whenever there is a change in the build. These enhancements would improve the
+state-of-the-art in IDE project import by not requiring users to manually
+check when the build and IDE have gone off-sync and ask for a "Reload
+Project". If sbt implemented BSP, an sbt shell would suffice to also support
```

Comment:
I think saying "If sbt implemented BSP" might make people wonder why you don't know about it. While that may be the reality, I think it would be better not to sound so distant from sbt's development, as this will be posted on the scala-lang.org blog, which aims to be authoritative. Maybe phrase this as, `In a future release of sbt supporting BSP, an sbt shell...`? Alternatives possible too. :)